mm2src/coins/eth/tron/address.rs

```diff
@@ -0,0 +1,151 @@
+//! TRON address handling (base58, hex, validation, serde).
+
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::convert::TryInto;
+use std::fmt;
+use std::str::FromStr;
+
+pub const ADDRESS_HEX_PREFIX: u8 = 0x41;
+pub const ADDRESS_BASE58_PREFIX: char = 'T';
+pub const ADDRESS_HEX_LEN: usize = 42;
+pub const ADDRESS_BYTES_LEN: usize = 21;
+pub const ADDRESS_BASE58_LEN: usize = 34;
+
+/// TRON mainnet or testnet address (21 bytes, 0x41 prefix + 20-bytes).
+#[derive(Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct Address {
+    pub inner: [u8; ADDRESS_BYTES_LEN],
+}
+
+impl Address {
+    /// Construct from raw 21 bytes (must be 0x41-prefixed).
+    pub fn from_bytes(bytes: [u8; ADDRESS_BYTES_LEN]) -> Result<Self, String> {
+        if bytes[0] != ADDRESS_HEX_PREFIX {
+            return Err("TRON address must start with 0x41".into());
+        }
+        Ok(Self { inner: bytes })
+    }
+
+    /// Construct from base58 string (with checksum).
+    pub fn from_base58(s: &str) -> Result<Self, String> {
+        let data = bs58::decode(s)
+            .with_check(None)
+            .into_vec()
+            .map_err(|e| format!("Invalid base58check address: {}", e))?;
+
+        if data.len() != ADDRESS_BYTES_LEN || data[0] != ADDRESS_HEX_PREFIX {
```

Comment:
nit: could u add a safety comment above this. not very trivial to see that such short circuit guarantee that we will not index an empty vector.
```rust
/// Index Safety: We will only index `data[0]` if data.len() == ADDRESS_BYTES_LEN != 0 (i.e. `data` vector won't be empty).

Reply:
ADDRESS_HEX_PREFIX is called _HEX_PREFIX even though it's used with base58 as well. we can just call it ADDRESS_PREFIX instead.

Reply:
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2425/commits/e9ba8d0e5226f5f290331b972d22c5c4fb97e3b8